### PR TITLE
Add documentation for `mesos_http_executor_domain_sockets`

### PR DESCRIPTION
**Pull request by bbannier**
## Jira Ticket
Before creating this pull request, please make sure you have a separate ticket for the docs team to track their work to review and merge, even for minor edits. If you need assistance, ask in the #doc-team channel on slack.

## Description of changes being made

- [D2IQ-61184](https://jira.d2iq.com/browse/D2IQ-61184)

## Checklist
- [x] Make sure you change all affected versions (e.g. 1.10, 1.11, 1.12, 1.13).
- [ ] Test all commands and procedures where applicable.
- [ ] Add [redirects](https://github.com/mesosphere/dcos-docs-site/wiki/Redirects) if you are moving a page.

See the [contribution guidelines](https://github.com/mesosphere/dcos-docs-site/wiki/Contributing) for more information.

### CODE REVIEW

**Comment by cmastr** [comment]
@bbannier I added a rough description to the release notes. Please review and tweak.

File: pages/mesosphere/dcos/2.1/release-notes/2.1.0/index.md [hunk 1]
```diff
@@ -22,100 +22,17 @@ This release provides new features and enhancements to improve the user experien
 
 If you have DC/OS deployed in a production environment, see [Known Issues and Limitations](#known-issues-and-limitations) to see if any potential operational changes for specific scenarios apply to your environment.
 
-- Updated DC/OS UI to master+v2.150.2.
```
Comment (bbannier) [inline pages/mesosphere/dcos/2.1/release-notes/2.1.0/index.md]: Why are you removing all of this if your commit message claims you only add something regarding domain sockets? This looks like an error to me.

Comment (cmastr) [reply pages/mesosphere/dcos/2.1/release-notes/2.1.0/index.md]: Sorry @bbannier maybe should have done a separate PR. This is our first release notes item for 2.1, so I needed to clean up the file for next release.

Comment (bbannier) [reply pages/mesosphere/dcos/2.1/release-notes/2.1.0/index.md]: Let's at least pull the cleanup into a separate commit with its own description. That would avoid any confusion whether the change was intentional or not.

**Comment by cmastr** [comment]
I reverted my changes to the Release Notes. We'll have to open a new PR to add release notes for this feature.

**Pull request merged**